### PR TITLE
Updates for 12.7.0

### PR DESCRIPTION
**Pull request by lilgreenbird**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,31 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/)
 
+## [12.7.0] Preview Release
+### Changed
+
+### Added
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:9]:
```suggestion
### Added
```

**Comment by Jeffery-Wasty** [comment]
Remove changes to README and changes to samples, we don't update those during a preview.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,31 @@ All notable changes to this project will be documented in this file.
...
+- Added support for TDSType.GUID [#2370] (https://github.com/microsoft/mssql-jdbc/pull/2370)
+- Allow constructing a microsoft.sql.DateTimeOffset instance from a java.time.OffsetDateTime value [#2340] (https://github.com/microsoft/mssql-jdbc/pull/2340)
+- Finish support for RFC4180 for CSV bulk insert operations [#2338] (https://github.com/microsoft/mssql-jdbc/pull/2338)
+- Server Message Handler and SQLException Chaining [#2251] (https://github.com/microsoft/mssql-jdbc/pull/2251)
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:13]: Sort old --> new instead of new --> old. See previous versions.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,31 @@ All notable changes to this project will be documented in this file.
...
+- Check if TDSCommand counter is null before incrementing. [#2368] (https://github.com/microsoft/mssql-jdbc/pull/2368)
+- Fixed ClassLoader leak of ActivityCorrelator ThreadLocal [#2366] (https://github.com/microsoft/mssql-jdbc/pull/2366)
+- Clear prepared statement cache when resetting statement pool connection [#2361] (https://github.com/microsoft/mssql-jdbc/pull/2361)
+- Clear prepared statement handle before reconnect [#2364] (https://github.com/microsoft/mssql-jdbc/pull/2364)
+- Reset socketTimeout to original value after a successful connection open [#2355] (https://github.com/microsoft/mssql-jdbc/pull/2355)
+- Fix calling procedures with output parameters by their four-part syntax [#2349] (https://github.com/microsoft/mssql-jdbc/pull/2349)
+- Added token cache map to fix use of unintended auth token for subsequent connections [#2341] (https://github.com/microsoft/mssql-jdbc/pull/2341)
+- Fix to ensure metadata returned follows JDBC data type specs [#2326] (https://github.com/microsoft/mssql-jdbc/pull/2326)
+- Fix to allow connection retries to be disabled by setting connectRetryCount to 0 [#2293] (https://github.com/microsoft/mssql-jdbc/pull/2293)
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:29]: Same as for "Added", these should be sorted with the oldest coming first.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,31 @@ All notable changes to this project will be documented in this file.
...
+-
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:30]:
```suggestion

```

**Comment by Jeffery-Wasty** [comment]
Also in the changelog, I added an extra hyphen last time on lines 40 and 41. Please remove.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,29 @@ All notable changes to this project will be documented in this file.
...
+- Remove synchronized from Socket overrides [#2337] (https://github.com/microsoft/mssql-jdbc/pull/2337)
+- Default to RMFAIL instead of RMERR [#2348] (https://github.com/microsoft/mssql-jdbc/pull/2348)
+
+### Added
+- Server Message Handler and SQLException Chaining [#2251] (https://github.com/microsoft/mssql-jdbc/pull/2251)
+- Finish support for RFC4180 for CSV bulk insert operations [#2338] (https://github.com/microsoft/mssql-jdbc/pull/2338)
+- Allow constructing a microsoft.sql.DateTimeOffset instance from a java.time.OffsetDateTime value [#2340] (https://github.com/microsoft/mssql-jdbc/pull/2340)
+- Added support for TDSType.GUID [#2370] (https://github.com/microsoft/mssql-jdbc/pull/2370)
+
+### Fixed issues
```
Comment (Jeffery-Wasty) [inline CHANGELOG.md:17]: Order should be "Added", "Changed", "Fixed issues"

**Review by tkyc — APPROVED** [review]

**Review by Jeffery-Wasty — APPROVED** [review]

**Pull request merged**